vm/interpreter/src/lib.rs

```diff
@@ -1,130 +1,318 @@
 // Copyright 2020 ChainSafe Systems
 // SPDX-License-Identifier: Apache-2.0, MIT
 
+use actor::{ActorState, ACCOUNT_ACTOR_CODE_ID, REWARD_ACTOR_ADDR};
 use address::Address;
 use blocks::Tipset;
 use clock::ChainEpoch;
-use default_runtime::{internal_send, DefaultRuntime};
+use default_runtime::{
+    internal_send, transfer_from_gas_holder, transfer_to_gas_holder, DefaultRuntime,
+};
+use forest_encoding::Cbor;
 use ipld_blockstore::BlockStore;
+use log::warn;
 use message::{Message, MessageReceipt, SignedMessage, UnsignedMessage};
 use num_bigint::BigUint;
-use num_traits::cast::ToPrimitive;
-use runtime::DefaultSyscalls;
-use vm::ActorError;
-use vm::{ExitCode, Serialized, StateTree};
-
+use num_traits::Zero;
+use runtime::Syscalls;
+use vm::{price_list_by_epoch, ActorError, ExitCode, Serialized, StateTree};
 /// Interpreter which handles execution of state transitioning messages and returns receipts
 /// from the vm execution.
-pub struct VM<'a, ST: StateTree, DB: BlockStore> {
+pub struct VM<'a, ST: StateTree, DB: BlockStore, SYS: Syscalls>
+where
+    SYS: Copy,
```

Comment:
just move the syscalls trait definition below or put both above, this is hard to follow (trait definitions in the where seem to make more sense to me personally)